File: Codigo_Fonte_pascal/lista2/lista2_6.pas
program chute;

var
   cont, input, valor : integer;

begin
     valor := 58;

     //writeln('Estrutura de repetição while...do');
     input := 0;
     cont := 0;

     while input <> valor do
     begin
          write('Digite um numero:');
          readln(input);
          if input < valor then
             writeln('MAIOR!');
          if input > valor then
             writeln('MENOR');
          cont := cont + 1;
     end;
     writeln('PARABENS, numero de tentativa: ', cont);
     writeln;

     //writeln('Estrutura de repetição repeat...until');
     input := 0;
     cont := 0;
     repeat
           write('Digite um numero:');
           readln(input);
           if input < valor then
              writeln('MAIOR!');
           if input > valor then
              writeln('MENOR');
           cont := cont + 1;
     until input = valor;
     writeln('PARABENS, numero de tentativa: ', cont);

     readln;

end.

File: Codigo_Fonte_c/Lista 2/lista2_6.c
#include <stdio.h>
#include <stdlib.h>

int main() {

	int contador, input, valor;
	valor = 54;
	
// estrutura de repetição while
	contador = 0;
	while (input != valor){
		printf("Digite um numero: ");
		scanf("%d", &input);
		if (input < valor){
			printf("MAIOR!\n");			
		}
		if (input > valor){
			printf("MENOR!\n");
		}
		contador++;
	}
	printf("PARABENS!, numero de tentativa: %d\n", contador);


// estrutura de repetição do while
	contador = 0;
	do
	{
		printf("Digite um numero: ");
		scanf("%d", &input);
		if (input < valor){
			printf("MAIOR!\n");			
		}
		if (input > valor){
			printf("MENOR!\n");
		}
		contador++;
	}while (input != valor);
	printf("PARABENS!, numero de tentativa: %d\n", contador);

  return 0;
}
